Services/Certificate/classes/Placeholder/Values/class.ilCoursePlaceholderValues.php

```diff
@@ -156,9 +178,23 @@ public function getPlaceholderValuesForPreview(int $userId, int $objId): array
 
         $placeholders = array_merge($placeholders, $customUserFieldsPlaceholders);
 
+        $lng_code = ilObjUser::_lookupLanguage($userId);
         $object = $this->objectHelper->getInstanceByObjId($objId);
+        $title = $object->getTitle();
+
+        $sql =
+            'SELECT title' . PHP_EOL
+            . 'FROM object_translation' . PHP_EOL
+            . 'WHERE obj_id = ' . $this->db->quote($objId, 'integer') . PHP_EOL
+            . 'AND lang_code = ' . $this->db->quote($lng_code, 'text') . PHP_EOL;
+        $r = $this->db->query($sql);
+        $row = $r->fetchRow(ilDBConstants::FETCHMODE_OBJECT);
+
+        if ($row) {
+            $title = (string) $row->title;
+        }
```

Comment:
Do we really need this database query? IMO it should be sufficient to rely on `ilObject` (and maybe additionally on `ilObjectTranslation`.